android/app/src/main/java/app/organicmaps/widget/menu/NavMenu.java

```diff
@@ -211,9 +214,20 @@ private void updateSpeedView(@NonNull RoutingInfo info)
 
     Pair<String, String> speedAndUnits = StringUtils.nativeFormatSpeedAndUnits(last.getSpeed());
 
+    if (info.speedLimitMps > 0.0)
+    {
+      Pair<String, String> speedLimitAndUnits = StringUtils.nativeFormatSpeedAndUnits(info.speedLimitMps);
+      mSpeedValue.setText(speedAndUnits.first + "\u202F/\u202F" + speedLimitAndUnits.first);
+    }
+    else
+      mSpeedValue.setText(speedAndUnits.first);
+
+    if (info.isSpeedLimitExceeded())
+      mSpeedValue.setTextColor(ContextCompat.getColor(mActivity, R.color.base_red));
+    else
+      mSpeedValue.setTextColor(ThemeUtils.getColor(mActivity, android.R.attr.textColorPrimary));
+
     mSpeedUnits.setText(speedAndUnits.second);
-    mSpeedValue.setText(speedAndUnits.first);
-    mSpeedViewContainer.setActivated(info.isSpeedLimitExceeded());
```

Comment:
This line made speedometer backgound red. Removed it. Now only text color is red.

Reply:
This comment is not relevant anymore, right?